src/ray/raylet/worker_pool.h

```diff
@@ -417,6 +457,24 @@ class WorkerPool : public WorkerPoolInterface, public IOWorkerPoolInterface {
   /// Get the NodeID of this worker pool.
   const NodeID &GetNodeID() const;
 
+  /// Returns a worker id if an idle worker is reused for this request. Otherwise,
+  /// asynchronously start a worker and return WorkerID::Nil().
+  WorkerID PopWorker(std::shared_ptr<PopWorkerRequest> pop_worker_request);
+
+  // Find an idle worker that can serve the task. If found, pop it out and return it.
+  // Otherwise, return nullptr.
+  std::shared_ptr<WorkerInterface> FindAndPopIdleWorker(
+      const PopWorkerRequest &pop_worker_request);
+
+  // Starts a new worker that fulfills `pop_worker_request`. Difference on methods:
+  // - PopWorker may reuse idle workers.
+  // - StartNewWorker force starts a new worker, with runtime env created.
+  // - StartWorkerProcess starts a new worker process, *without* runtime env creation.
+  //
+  // Note: NONE of these methods guarantee that pop_worker_request.callback will be called
+  // with the started worker. It may be called with any fitting workers.
+  void StartNewWorker(const std::shared_ptr<PopWorkerRequest> &pop_worker_request);
```

Comment:
`PrestartWorker`?

Reply:
Because this is also used internally by PopWorker